### ISSUE TITLE
Unicode-escaped national characters are shown instead of human readable text

### ISSUE THREAD

**Issue by another-guy** [reactjs/reactjs.org#1732]
The issue is migrated from the ru.reactjs.org: https://github.com/reactjs/ru.reactjs.org/issues/180

@gcor noticed a bug in the way national characters are displayed in Code Editor **with JSX checkbox disabled (reset)**. **Expected** to see human-readable text in a particular natural language (when used in string literals and comments). **Actually** seeing unicode-escaped symbols. (See images below). 

At least two languages are known to be affected: Russian and Armenian, but the issue must be not specific to just these two.

**Russian translation**

<img width="606" alt="screen shot 2019-02-22 at 12 35 12 pm" src="https://user-images.githubusercontent.com/4715396/53233458-870cb380-369e-11e9-8269-7e77e3460a74.png">
<img width="589" alt="screen shot 2019-02-22 at 12 35 19 pm" src="https://user-images.githubusercontent.com/4715396/53233459-870cb380-369e-11e9-9c23-312838ca46da.png">

**Armenian translation**

![image](https://user-images.githubusercontent.com/5762340/53234288-1042cc80-3644-11e9-8f81-60f2c15a896d.png)

**Comment by tesseralis** [comment]
It seems to be an issue with Babel. When you uncheck the checkbox it runs Babel to compile it to es5. For some reason it also escapes unicode characters, and I can't figure out the setting to fix it.

@gaearon do you know who would know more about babel presets/plugins?

**Comment by tesseralis** [comment]
Ooh: https://github.com/babel/babel/issues/4909

**Comment by nutboltu** [comment]
If we try `<h3>{'আউটপুট'}</h3>` then it works

**Comment by another-guy** [comment]
@nutboltu having such code in documentation samples complicates things unnecessarily. The reader will get confused about why it's written that way, whether it's a pattern, or worse -- decide to use it across their project.

So, the issue should be fixed without forcing us to make code changes in examples ... Ideally.

**Comment by tesseralis** [comment]
As a stopgap solution we could just keep the text on the code examples in English (that's what some languages do anyways). @another-guy would you have the time to look into possible solutions?

**Comment by tesseralis** [comment]
The `jsonCompatibleStrings` option here might help: https://babeljs.io/docs/en/babel-generator

Unfortunately I'm being kicked out of this cafe so I can't try it out right now.

**Comment by another-guy** [comment]
@tesseralis I'm sorry, at the moment I can only focus on translating and pull request reviews. (I have a limiting schedule: busy with young daughter and full time job [which does not reserve time for OSS contribution]). I hope, you can understand me.

P.S. Sorry to hear you were kicked out of a cafe. This is crazy _!_

**Comment by tesseralis** [comment]
@another-guy yeah, that's fair. I'll look into it when I have time but I'm moving on to other projects right now :)

(also I should have specified that I got kicked out because the cafe was closing!)

**Comment by nutboltu** [comment]
@tesseralis and @another-guy 

I tried this in `CodeEditor.js`
```
const compileES6 = code => Babel.transform(code, {
  presets: ['react'],
  generatorOpts: {
    jsescOption: {
      minimal: true,
    },
  },
}).code;
```
It didn't work.

I also upgrades babel standalone version from `6.26.0` to `7.0.0-beta.3`.
In both versions, when I tried transform in the console, I couldn't find such `generatorOpts` options.

<img width="742" alt="babel_transform" src="https://user-images.githubusercontent.com/9896958/53599521-7b1f7500-3bfb-11e9-9ce8-81071de6b000.png">

Any thoughts?
